File: programmers/42586/solution.py
import math


def solution(progresses, speeds):
    answer = []
    working_days = [
        math.ceil((100 - p) / s)
        for p, s in zip(progresses, speeds)
    ]
    q = 0
    for i in range(len(working_days)):
        if working_days[i] > working_days[q]:
            answer.append(i - q)
            q = i
    answer.append(len(working_days) - q)
    return answer


print(solution([93, 30, 55], [1, 30, 5]))

File: programmers/42586/solution.js
function solution(progresses, speeds) {
  var answer = [];
  var day = 0
  var count = 0
  progresses.map((v,i) => {
      if (Math.ceil((100 - v)/speeds[i]) > day) {
          answer.push(count)
          day = Math.ceil((100 - v)/speeds[i]) 
          count = 1
          
      } else {
          count++
      }
  })
  answer.push(count)
  return answer.slice(1);
}

console.log(solution([93, 30, 55], [1, 30, 5])) // [2, 1]